javascript/javascript3/week3/TASK3-Shopping cart/index.js

```diff
@@ -1,149 +1,142 @@
-      }
-    }
+  getTotal() {
+    return this.products.reduce((result, currentProduct) => result + currentProduct.price, 0);
+  }
 
-    getUser = async (userID) => {
-      try {
-        const baseUrl = "https://jsonplaceholder.typicode.com/users/";
-        const query = userID;
-        return await getData(baseUrl, query);
-      } catch (error) {
-        console.log(error);
-      }
-    };
+  renderProducts() {
+    const fragment = new DocumentFragment();
+    const ul = document.createElement("UL");
+    fragment.appendChild(ul);
+    this.products.forEach((product) => {
+      createListElement(product);
+    });
   }
 
-  getData = async (baseUrl, query) => {
+  async getUser(userID) {
     try {
-      const response = await fetch(`${baseUrl}${query}`);
-      if (response.status >= 200 && response.status < 300) {
-        const data = response.json();
-        return data;
-      } else {
-        throw new Error(`HTTP error! status: ${response.status}`);
-      }
+      const BASE_URL = "https://jsonplaceholder.typicode.com/users/";
+      const query = userID;
+      return getData(BASE_URL, query);
     } catch (error) {
       console.log(error);
     }
-  };
-
-  async function getCurrencyRate(amountToConvert, convertToCurrency) {
-    const baseUrl = `https://v6.exchangerate-api.com/v6/84a875e2ff2c28d158651393/latest/`;
-    const query = defaultCurrency;
-    const data = await getData(baseUrl, query);
-    const currenciesArray = data.conversion_rates;
-    const convertedPrice = (amountToConvert * currenciesArray[`${convertToCurrency}`]).toFixed(2);
-    console.log(`${convertedPrice}  ${convertToCurrency}`);
-    return `${convertedPrice}  ${convertToCurrency}`;
   }
-
-  async function displayEverything() {
-    await displayUserInfo();
-    shoppingCart.renderProducts();
-    displayTotalPrice();
-  }
-
-  async function displayUserInfo() {
-    let user = await shoppingCart.getUser(1);
-    const fragment = new DocumentFragment();
-    const userName = document.createElement("H2");
-    userName.textContent = user.name;
-    fragment.appendChild(userName);
-    document.body.append(fragment);
-  }
-
-  function displayTotalPrice() {
-    const totalPrice = document.createElement("h2");
-    totalPrice.textContent = `Total: ${shoppingCart.getTotal()}`;
-    const totalPriceToConvert = shoppingCart.getTotal();
-    document.body.appendChild(totalPrice);
+}
+
+function createListElement(productToList) {
+  let mainLI = document.createElement("LI");
+  ul.appendChild(mainLI);
+  let productUL = document.createElement("UL");
+  mainLI.appendChild(productUL);
+  let nameLI = document.createElement("LI");
+  nameLI.textContent = `Procuct: ${productToList.name}`;
+  productUL.append(nameLI);
+  let priceLI = document.createElement("LI");
+  priceLI.textContent = `Price: ${productToList.price}`;
+  productUL.append(priceLI);
+  document.body.appendChild(fragment);
+}
+
+async function getData(BASE_URL, query) {
```

Comment:
just a nitpick, but `getData` is the least useful function name ever :) What _kind_ of data will this function get?